src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/TaskAutoAssignmentService.java

```diff
@@ -115,6 +95,44 @@ public TaskResource autoAssignCFTTask(TaskResource taskResource) {
         return taskResource;
     }
 
+    public boolean checkAssigneeIsStillValid(TaskResource taskResource, String assignee) {
+
+        List<RoleAssignment> roleAssignments = taskConfigurationRoleAssignmentService.getRolesByUserId(assignee);
+
+        Optional<RoleAssignment> match = runRoleAssignmentAutoAssignVerification(taskResource, roleAssignments);
+        return match.isPresent();
+    }
+
+    private Optional<RoleAssignment> runRoleAssignmentAutoAssignVerification(TaskResource taskResource,
+                                                                             List<RoleAssignment> roleAssignments) {
+        // the lowest assignment priority takes precedence.
+        List<TaskRoleResource> rolesList = new ArrayList<>(taskResource.getTaskRoleResources());
+        rolesList.sort(comparing(TaskRoleResource::getAssignmentPriority, nullsLast(Comparator.naturalOrder())));
+
+        Map<String, TaskRoleResource> roleResourceMap = rolesList.stream()
+            .collect(toMap(
+                TaskRoleResource::getRoleName,
+                taskRoleResource -> taskRoleResource,
+                (a, b) -> b
+            ));
+
+        List<RoleAssignment> orderedRoleAssignments = orderRoleAssignments(rolesList, roleAssignments);
+        return orderedRoleAssignments.stream()
+            .filter(roleAssignment -> {
+                TaskRoleResource taskRoleResource = roleResourceMap.get(roleAssignment.getRoleName());
+                if (taskRoleResource == null
+                    || taskRoleResource.getAuthorizations() == null
```

Comment:
can we add coverage for this -> taskRoleResource == null
                    || taskRoleResource.getAuthorizations() == null

Reply:
added